lib/plugins/posthtml-flatten-style.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+var url = require('url');
+var Bluebird = require('bluebird');
+var collapseCSS = require('../collapsers/css');
+
+module.exports = function (opts) {
+  return function posthtmlFlattenStyle(tree) {
+    return Bluebird.try(function () {
+      var tasks = [];
+
+      tree.match([
+        {
+          tag: 'link',
+          attrs: {
+            href: true,
+            rel: 'stylesheet'
+          }
+        },
+        {
+          tag: 'style'
+        }
+      ], function (node) {
```

Comment:
Can we add a matcher for `style` attributes as well to match e.g. `<div style="background-image: url(...)">`.